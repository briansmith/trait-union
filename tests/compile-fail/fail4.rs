use trait_union::trait_union;
use std::fmt::Debug;

trait_union! {
    union U<T>: Debug where T: Debug+Copy = T;
}

fn main() { }
