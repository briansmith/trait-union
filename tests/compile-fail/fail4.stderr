error[E0310]: the parameter type `T` may not live long enough
 --> $DIR/fail4.rs:5:11
  |
5 |     union U<T>: Debug where T: Debug+Copy = T;
  |           ^^-^
  |           | |
  |           | help: consider adding an explicit lifetime bound...: `T: 'static`
  |           ...so that the type `T` will meet its required lifetime bounds
